### PR TITLE
Registration: Video background

### PR DESCRIPTION
**Pull request by kirkas**

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/next.config.js [hunk 1]
```diff
@@ -159,6 +159,22 @@ module.exports = extendBaseConfig(
...
       defaultLocale: 'en',
     },
+    webpack: (config, { buildId, dev, isServer, defaultLoaders, nextRuntime, webpack }) => {
+      config.module.rules.push({
+        test: /\.webm/,
+        use: [
+          {
+            loader: 'file-loader',
+            options: {
+              name: '[name].[hash].[ext]',
```
Comment (JFrankfurt) [inline apps/web/next.config.js:169]:
```suggestion
              name: '[name][hash].[ext]',
```
nit

File: apps/web/src/components/Basenames/RegistrationBackground/index.tsx [hunk 2]
```diff
@@ -16,14 +20,16 @@ export default function RegistrationBackground() {
   const isPending = registrationStep === RegistrationSteps.Pending;
   const isSuccess = registrationStep === RegistrationSteps.Success;
 
-  const claimBackgroundClasses = classNames(
-    'pointer-events-none absolute inset-0 w-full h-full bg-cover bg-center -z-10',
```
Comment (JFrankfurt) [inline apps/web/src/components/Basenames/RegistrationBackground/index.tsx:20]: double checking you intend to remove the negative from z-10 here

File: apps/web/src/components/Basenames/RegistrationBackground/index.tsx [hunk 3]
```diff
@@ -39,31 +45,77 @@ export default function RegistrationBackground() {
...
+        leaveFrom="opacity-100"
+        leaveTo="opacity-0"
+      >
+        {/* TODO: Lottie animation file */}
+
+        <Transition
+          appear
+          show={isClaim}
+          className={classNames(
+            'transform-gpu transition-all',
```
Comment (JFrankfurt) [inline apps/web/src/components/Basenames/RegistrationBackground/index.tsx:65]: `transform-gpu` is new to me! v cool

File: apps/web/src/components/Basenames/RegistrationBackground/index.tsx [hunk 4]
```diff
@@ -39,31 +45,77 @@ export default function RegistrationBackground() {
...
+            registrationTransitionDuration,
+            videoClasses,
+          )}
+          enterFrom="scale-0"
+          enterTo="scale-100"
+        >
+          {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
```
Comment (JFrankfurt) [inline apps/web/src/components/Basenames/RegistrationBackground/index.tsx:72]: lol captions

File: apps/web/src/components/Basenames/RegistrationBackground/index.tsx (hunk 2 above)
Comment (kirkas) [reply apps/web/src/components/Basenames/RegistrationBackground/index.tsx:20]: yeah it's on these classes

grayVideoBackgroundClasses
blueVideoBackgroundClasses

File: apps/web/src/components/Basenames/RegistrationBackground/index.tsx (hunk 4 above)
Comment (kirkas) [reply apps/web/src/components/Basenames/RegistrationBackground/index.tsx:72]: 🙃

File: apps/web/src/components/Basenames/RegistrationBackground/index.tsx [hunk 5]
```diff
@@ -39,31 +45,77 @@ export default function RegistrationBackground() {
...
+          <video autoPlay className={videoClasses} loop>
+            <source src={globe} type="video/webm" />
```
Comment (JFrankfurt) [inline apps/web/src/components/Basenames/RegistrationBackground/index.tsx:74]: consider adding a fallback mp4 source (5% of global devices are on a version of ios that only has partial support for webm)
https://caniuse.com/webm

**Review by JFrankfurt — APPROVED** [review]

File: apps/web/src/components/Basenames/RegistrationBackground/index.tsx (hunk 3 above)
Comment (kirkas) [reply apps/web/src/components/Basenames/RegistrationBackground/index.tsx:65]: yeah it's nice to leverage GPU, tho API for it is not always supported (fallback to cpu)

File: apps/web/src/components/Basenames/RegistrationBackground/index.tsx (hunk 5 above)
Comment (kirkas) [reply apps/web/src/components/Basenames/RegistrationBackground/index.tsx:74]: fallback is not video (just gray or blue background) - I wanted to add mp4 but the files sizes are too large IMO (11mb for fireworks.mp4)

**Pull request merged**